test/factories/host_related.rb

```diff
@@ -12,6 +12,34 @@ def set_nic_attributes(host, attributes, evaluator)
   host
 end
 
+# @param subnet [Optional[Subnet::Ipv4]]
+# @param n [Integer]
+# @param base [Integer]
+#   The base address if subnet is nil
+def ip_from_subnet(subnet, n = 1, base = 0)
+  addr = if (ipaddr = subnet&.ipaddr)
+           available_ips = 2**(32 - ipaddr.prefix) - 1
```

Comment:
I considered but then you still need the family in `IPAddr.new` in case `subnet` is nil.
```suggestion
           available_ips = 2**((subnet.family == Socket::AF_INET ? 32 : 128) - ipaddr.prefix) - 1
```

And then the `- 1` is technically not needed for IPv6, but I thought that there's not as much benefit of abstracting IPv4 & IPv6 here.

Reply:
I agree.